src/reactors/V2DutchOrderReactor.sol

```diff
@@ -0,0 +1,133 @@
+        virtual
+        override
+        returns (ResolvedOrder memory resolvedOrder)
+    {
+        V2DutchOrder memory order = abi.decode(signedOrder.order, (V2DutchOrder));
+        // hash the order _before_ overriding amounts, as this is the hash the user would have signed
+        bytes32 orderHash = order.hash();
+
+        _validateOrder(orderHash, order);
+        _updateWithOverrides(order);
+
+        resolvedOrder = ResolvedOrder({
+            info: order.info,
+            input: order.input.decay(order.cosignerData.decayStartTime, order.cosignerData.decayEndTime),
+            outputs: order.outputs.decay(order.cosignerData.decayStartTime, order.cosignerData.decayEndTime),
+            sig: signedOrder.sig,
+            hash: orderHash
+        });
+    }
+
+    /// @inheritdoc BaseReactor
+    function transferInputTokens(ResolvedOrder memory order, address to) internal override {
+        permit2.permitWitnessTransferFrom(
+            order.toPermit(),
+            order.transferDetails(to),
+            order.info.swapper,
+            order.hash,
+            V2DutchOrderLib.PERMIT2_ORDER_TYPE,
+            order.sig
+        );
+    }
+
+    function _updateWithOverrides(V2DutchOrder memory order) internal pure {
+        if (order.cosignerData.inputOverride != 0) {
+            if (order.cosignerData.inputOverride > order.input.startAmount) {
+                revert InvalidInputOverride();
+            }
+            order.input.startAmount = order.cosignerData.inputOverride;
+        }
+
+        if (order.cosignerData.outputOverrides.length != order.outputs.length) {
+            revert InvalidOutputOverride();
+        }
+        for (uint256 i = 0; i < order.outputs.length; i++) {
+            DutchOutput memory output = order.outputs[i];
+            uint256 outputOverride = order.cosignerData.outputOverrides[i];
+            if (outputOverride != 0) {
+                if (outputOverride < output.startAmount) {
+                    revert InvalidOutputOverride();
+                }
+                output.startAmount = outputOverride;
+            }
+        }
+    }
+
+    /// @notice validate the dutch order fields
+    /// - deadline must be greater than or equal than decayEndTime
+    /// - decayEndTime must be greater than or equal to decayStartTime
+    /// - if there's input decay, outputs must not decay
+    /// @dev Throws if the order is invalid
+    function _validateOrder(bytes32 orderHash, V2DutchOrder memory order) internal pure {
+        if (order.info.deadline < order.cosignerData.decayEndTime) {
+            revert DeadlineBeforeEndTime();
+        }
+
+        if (order.cosignerData.decayEndTime <= order.cosignerData.decayStartTime) {
+            revert DutchDecayLib.EndTimeBeforeStartTime();
+        }
+
+        (bytes32 r, bytes32 s) = abi.decode(order.cosignature, (bytes32, bytes32));
+        uint8 v = uint8(order.cosignature[64]);
+        // cosigner signs over (orderHash || cosignerData)
```

Comment:
this means `orderHash` concat `cosignerData` right? at first I thought it's logical or

Reply:
oh yea its concat

Reply:
i was taught at university that this was the symbol for concat... but now that im googling i think that ive just been living a lie 😆 
![image](https://github.com/Uniswap/UniswapX/assets/34962750/d2bd292e-6137-42e5-81cd-35de7795ceeb)